analyses/copy_number_consensus_call/src/scripts/compare_variant_calling_updated.py

```diff
@@ -0,0 +1,257 @@
+            ## ASSUMING there is no overlaping segments within 1 method
+            ## then it is safe to add/accumulate the coverage_list1 value. Refer to assumptions (line 23)
+
+            ## Next we calculate the coverage of list2 since there might be more than 1 overlapping CNV from list2
+            if list2_total_len > 0:
+                coverage_list2 = list2_overlap_len / list2_total_len
+            else:
+                coverage_list2 = 0
+
+            ## Check to see if the overlap is >= 50% RECIPROCALLY
+            if coverage_list1 >= 0.5 and coverage_list2 >= 0.5:
+                ## if it is 50% reciprocally, we chose from the list2 starts and ends
+                ## we take out the smallest start and biggest end
+                ## This maximize the CNV length from list2
+                fin_start_list2 = min(list2_start_list)
+                fin_end_list2 = max(list2_end_list)
+
+                ## Taking only the common region as the final segment.
+                ## Compare between list2 vs list1 start and end - take the common segment
+                chrom_start = max(fin_start_list2,start_list1)
+                chrom_end = min(fin_end_list2,end_list1)
+
+                ## Format the output consensus CNV
+                ## Column 4 is manta's raw coordinates
+                ## Column 5 is cnvkit's raw coordinates
+                ## Column 6 is freec's raw coordinates
+
+                ## if jval == 0 (manta) and kval == 1 (cnvkit), put info in column 4 and 5, 6th column is null
+                if jval == 0 and kval == 1:
+                    overlap_chrom = [m[0],str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),list2_chr_str_end.strip(','),'NULL',m[-1]]
+
+                ## if jval == 0 (manta) and kval == 2 (freec), put info in column 4 and 6, 5th column is null
+                elif jval == 0 and kval == 2:
+                    overlap_chrom = [m[0],str(chrom_start),str(chrom_end),str(cnv_list1).strip(','),'NULL',list2_chr_str_end.strip(','),m[-1]]
+
+                ## if jval == 1 (cnvkit) and kval == 2 (freec), put info in column 5 and 6, 4th column is null
+                elif jval == 1 and kval == 2:
+                    overlap_chrom = [m[0],str(chrom_start),str(chrom_end),'NULL',str(cnv_list1).strip(','),list2_chr_str_end.strip(','),m[-1]]
+
+                ## Add the formatted consensus CNV into a growing list of consensus CNVs
+                consensus_list = consensus_list + [overlap_chrom]
+
+        ## Add the consensus list into a final list that hold the
+        ## content of 3 files - manta_cnvkit , manta_freec, and cnvkit_freec
+        fin_list += [consensus_list]
+
+## Print the output into files
+for i,ival in enumerate(fin_list):
+    try:
+        ## Attempt to create a the file, if the file already exists, throw ERROR
+        with open( list_of_output_files[i] , 'x') as file:
+            sys.stderr.write('$$$ Created new file succesfully\n')
```

Comment:
I just made a few small files and did a test. It seems that Snakemake DOES delete the output file, then regenerate that file if the input is at a later time stamp than the output. 

I put this step in so that the script doesn't override any info that the user might not want to be deleted. But I think in the context of this pipeline, we could just add a step to override the output file. What do you think?

Reply:
In that case I don't think it matters much. I would personally have an option for overwriting, but I don't feel strongly.